packages/material-ui/src/Slider/Slider.test.js

```diff
@@ -0,0 +1,470 @@
+        changedTouches: touchList([{ identifier: 1, pageX: 22, pageY: 0 }]),
+      });
+
+      expect(handleChange.callCount).to.equal(3);
+      expect(handleChange.args[0][1]).to.deep.equal([21, 30]);
+      expect(handleChange.args[1][1]).to.deep.equal([22, 30]);
+      expect(handleChange.args[2][1]).to.equal(handleChange.args[1][1]);
+    });
+  });
+
+  describe('prop: step', () => {
+    it('should handle a null step', () => {
+      const { getByRole, container } = render(
+        <Slider
+          step={null}
+          marks={[{ value: 0 }, { value: 20 }, { value: 30 }]}
+          defaultValue={0}
+        />,
+      );
+      stub(container.firstChild, 'getBoundingClientRect').callsFake(() => ({
+        width: 100,
+        height: 10,
+        bottom: 10,
+        left: 0,
+      }));
+      const thumb = getByRole('slider');
+
+      const event = fireBodyMouseEvent('touchstart', {
+        changedTouches: touchList([{ identifier: 1, pageX: 21, pageY: 0 }]),
+      });
+      container.firstChild.dispatchEvent(event);
+      expect(thumb.getAttribute('aria-valuenow')).to.equal('20');
+
+      fireEvent.keyDown(thumb, {
+        key: 'ArrowUp',
+      });
+      expect(thumb.getAttribute('aria-valuenow')).to.equal('30');
+
+      fireEvent.keyDown(thumb, {
+        key: 'ArrowDown',
+      });
+      expect(thumb.getAttribute('aria-valuenow')).to.equal('20');
+    });
+  });
+
+  describe('prop: disabled', () => {
+    it('should render the disabled classes', () => {
+      const { container } = render(<Slider disabled value={0} />);
+      expect(container.firstChild).to.have.class(classes.disabled);
+    });
+  });
+
+  describe('keyboard', () => {
+    it('should handle all the keys', () => {
+      const { getByRole } = render(<Slider defaultValue={50} />);
+      const thumb = getByRole('slider');
+
+      fireEvent.keyDown(thumb, {
+        key: 'Home',
+      });
+      expect(thumb.getAttribute('aria-valuenow')).to.equal('0');
+
+      fireEvent.keyDown(thumb, {
+        key: 'End',
+      });
+      expect(thumb.getAttribute('aria-valuenow')).to.equal('100');
+
+      fireEvent.keyDown(thumb, {
+        key: 'PageDown',
+      });
+      expect(thumb.getAttribute('aria-valuenow')).to.equal('90');
+
+      fireEvent.keyDown(thumb, {
+        key: 'Escape',
+      });
+      expect(thumb.getAttribute('aria-valuenow')).to.equal('90');
+
+      fireEvent.keyDown(thumb, {
+        key: 'PageUp',
+      });
+      expect(thumb.getAttribute('aria-valuenow')).to.equal('100');
+    });
+
+    const moveLeftEvent = {
+      key: 'ArrowLeft',
+    };
+    const moveRightEvent = {
+      key: 'ArrowRight',
+    };
+
+    it('should reach right edge value', () => {
+      const { getByRole } = render(<Slider defaultValue={90} min={6} max={108} step={10} />);
+      const thumb = getByRole('slider');
+
+      fireEvent.keyDown(thumb, moveRightEvent);
+      expect(thumb.getAttribute('aria-valuenow')).to.equal('100');
```

Comment:
```suggestion
      expect(thumb).to.have.attribute('aria-valuenow', '100');
```

Reply:
- pros: you get the node printed when the assertion fails
- cons: you have to learn a new API

Is that it?

Reply:
> you have to learn a new API

I'm tired of this argument. It's not true. You have to learn both. In one case DOM the other case the assertion API. They both encode the same. Would appreciate it you stop spreading this misconception or make it clear that you have to learn it i.e. "I have to learn a new API" is what you mean.

Reply:
> pros: you get the node printed when the assertion fails

This is incomplete (surprise). It also prints what property you used (`aria-valuenow`) and it prints the value whereas the previous example just prints `expected not equal to false`

Reply:
Let's take an average developer, how much likely does it already know the DOM API vs the chai-dom API? How much value does it get, in the long run, at learning one of these two APIs? I have some concern with the nonpopularity of this approach:
- https://npm-stat.com/charts.html?package=chai&package=chai-dom
- https://npm-stat.com/charts.html?package=jest&package=jest-dom&from=2018-07-02&to=2019-07-02

There is no question the test fail output is better. Does is worth the tradeoff? 
This topic wasn't discussed https://github.com/mui-org/material-ui/pull/15732.

Now, it seems that random contributors rarely write new tests. We should probably not optimize for them. So I think that it's fine. Let's use it.

Reply:
Please show me the different API. There's different syntax, sure. Are you actually making an argument that this is difficult? Are we having a serious discussion or are you just justifying starting an argument over and over?

> Let's take an average developer, how much likely does it already know the DOM API vs the chai-dom API?

Neither. + it's not "chai-dom" API. It's common expect style.

But yeah I'm not arguing anymore since you clearly have no interest in changing your mind (proofen by repeating rebutted arguments with confidence in their self-evidence).

> This topic wasn't discussed #15732.

Yeah we're having no argument here. You're arguing in bad faith and I'm not wasting my time with this.

Reply:
> There's different syntax, sure. Are you actually making an argument that this is difficult?

Yes, this is the root of my argument. The concept is identical, the API/syntax is different. I believe that the difference is whether or not you have to go to the documentation, there is a learning curve.

It's a problem I have faced a while back when entering a new codebase. I couldn't figure out where these expect APIs were coming from. At that time, I wished it was using the web API.
The backend developers were struggling.